tests/test_viewers.py

```diff
@@ -0,0 +1,63 @@
+import pytest
+from aiida import engine, orm
+
+
+@pytest.mark.usefixtures("aiida_profile_clean")
+def test_several_data_viewers(bands_data_object, folder_data_object):
+    from aiidalab_widgets_base import viewer, viewers
+
+    v = viewer(orm.Int(1))
+
+    # No viewer for Int, so it should return the input
+    assert isinstance(v, orm.Int)
+
+    # DictViewer
+    v = viewer(orm.Dict(dict={"a": 1}))
+    assert isinstance(v, viewers.DictViewer)
+
+    # BandsDataViewer
+    v = viewer(bands_data_object)
+    assert isinstance(v, viewers.BandsDataViewer)
+
+    # FolderDataViewer
+    v = viewer(folder_data_object)
+    assert isinstance(v, viewers.FolderDataViewer)
+
+    # ProcessNodeViewer
+    from aiida.workflows.arithmetic.add_multiply import add_multiply
+
+    result, workfunction = engine.run_get_node(
+        add_multiply, orm.Int(3), orm.Int(4), orm.Int(5)
+    )
+    v = viewer(workfunction)
+    assert isinstance(v, viewers.ProcessNodeViewerWidget)
+
+
+@pytest.mark.usefixtures("aiida_profile_clean")
+def test_structure_data_viwer(structure_data_object):
+    from aiidalab_widgets_base import viewer, viewers
+
+    v = viewer(structure_data_object)
+    assert isinstance(v, viewers.StructureDataViewer)
+
+    # Selection of atoms.
+
+    # Direct selection
```

Comment:
Can you please test what happens when the selection is invalid. Specifically, try selecting atom 0, or atom that is greater then a number of atoms. I noticed a while ago that there might be bugs...

Reply:
Actually, that is fixed in #373. I added a note to test it there: https://github.com/aiidalab/aiidalab-widgets-base/pull/373#issuecomment-1472235131